lightning/src/ln/channel.rs

```diff
@@ -2665,14 +2665,16 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 		let proposed_total_fee_satoshis = proposed_feerate * tx_weight / 1000;
 
 		let (closing_tx, total_fee_satoshis) = self.build_closing_transaction(proposed_total_fee_satoshis, false);
-		let funding_redeemscript = self.get_funding_redeemscript();
-		let sighash = hash_to_message!(&bip143::SighashComponents::new(&closing_tx).sighash_all(&closing_tx.input[0], &funding_redeemscript, self.channel_value_satoshis)[..]);
+		let our_sig = self.local_keys
```

Comment:
Something like `if let Ok(our_sig) { Some() } else { None }` would avoid conversion to option and calling unwrap. Would make code easier to review too.

Reply:
Ehh, I dont think so? map_err()? is pretty clear.